src/var.c

```diff
@@ -250,11 +269,58 @@ static int fi_parse_bool(const char *str_value)
 	return -1;
 }
 
+static void load_conf() {
+	char *line = NULL;
+	char *tmp;
+	struct ofi_conf_entry *conf;
+	size_t length;
+	ssize_t read;
+
+	dlist_init(&conf_list);
+
+	FILE *fptr = fopen(DEFAULT_CONF_FILE_PATH, "r");
+	if (!fptr)
+		return;
+
+	while ((read = getline(&line, &length, fptr)) != -1) {
+		conf = calloc(1, sizeof(*conf));
+		if (!conf)
+			break;
+
+		tmp = strchr(line, '=');
+		tmp[0] = 0;
+		conf->var_name = strdup(line);
+		if (line[read-1] == '\n')
+			line[read-1] = 0;
+		conf->value = strdup(&tmp[1]);
+		dlist_insert_tail(&conf->entry, &conf_list);
+	}
+
+	free(line);
+	fclose(fptr);
+}
+
+static struct ofi_conf_entry *
+find_conf_entry(const struct fi_provider *provider, const char *param_name)
```

Comment:
`provider` is only used for debugging output. suggest to remove it from the parameter and move the debug output to the caller.